docs/cryptodoc/src/05_09_kyber.rst

```diff
@@ -141,161 +302,115 @@ Based on these functions the key generation process follows **Algorithms 4 and 7
 
    **Steps:**
 
-   1. Generate the random seed ``d`` and the implicit rejection value ``z`` at random using ``rng``
-   2. ``(rho || sigma) = G(d)`` (L. 1-2, Alg. 4 [Kyber-R3]_)
-   3. ``a = PolynomialMatrix::generate(rho, false, m)`` (``false`` means "not transposed") (L. 4-8, Alg. 4 [Kyber-R3]_)
-   4. ``s = PolynomialVector::getnoise_eta1(sigma, 0, m)`` (performs ``k`` invocations of ``Polynomial::getnoise_eta1``, one for each component of ``s``; L. 9-12, Alg. 4 [Kyber-R3]_)
-   5. ``e = PolynomialVector::getnoise_eta1(sigma, k, m)`` (performs ``k`` invocations of ``Polynomial::getnoise_eta1``, one for each component of ``e``; L. 13-16, Alg. 4 [Kyber-R3]_)
-   6. ``s.ntt()`` and ``e.ntt()`` (L. 17-18, Alg. 4 [Kyber-R3]_)
-   7. ``pk = (a*s + e, rho)`` and ``sk = (s, z)`` (L. 19-22, Alg. 4 [Kyber-R3]_ and L.1, 3, Alg. 7 [Kyber-R3]_)
+   1. Generate the random seed ``seed.d`` and the implicit rejection value ``seed.z`` at random using ``rng``
+   2. ``(rho, sigma) = G(d)``
+   3. Sample matrix ``A`` from ``rho`` using ``sample_matrix``
+   4. Initialize a ``PolynomialSampler`` ``ps`` with ``sigma``
+   5. ``s = ntt(ps.sample_polynomial_vector_cbd_eta1())``
+   6. ``e = ntt(ps.sample_polynomial_vector_cbd_eta1())``
+   7. Compute ``t = A * s + e``
+   8. ``pk = (t, rho)`` and ``sk = (seed.d, seed.z)``
 
    **Notes:**
 
-   - The member function ``Polynomial::getnoise_eta1(seed, nonce, mode)`` uses ``PRF`` on the seed with incremented nonce values to call ``Polynomial::getnoise_cbd2`` or ``Polynomial::getnoise_cbd3`` depending on ``eta1``.
-   - Serialization to bytes of the keys (:math:`\mathsf{Encode}` in L.20, 21, Alg. 4 [Kyber-R3]_) is performed via the constructor of the internal classes for public and secret keys (``Kyber_PublicKeyInternal`` and ``Kyber_PrivateKeyInternal``) by calling ``Polynomial::to_bytes()``.
+   - Step 1 corresponds to Algorithm 19 of [FIPS-203]_ and is performed in :srcref:`[src/lib/pubkey/kyber]/kyber/kyber_common/kyber.cpp:232|Kyber_PrivateKey::Kyber_PrivateKey`.
+   - Steps 2-7 correspond to Algorithms 16 and 13 of [FIPS-203]_ and are performed in :srcref:`[src/lib/pubkey/kyber]/kyber/kyber_common/kyber_algos.cpp:321|expand_keypair`.
+   - Botan only stores the seeds as the secret key. The required values for decapsulation are recomputed on demand.
```

Comment:
```suggestion
   - Botan only stores the seeds as the secret key. The required values for
     decapsulation are recomputed on demand. Loading or storing the partially
     expanded key format specified in [FIPS-203]_ is explicitly not supported.
```

Reply:
r

Reply:
Maybe reference the consensus in the LAMPS WG on this? https://mailarchive.ietf.org/arch/msg/spasm/KPg3lbPeNms2H5PNDBt0iJByuE0/

Reply:
Good point! I'll add a footnote.